// .ignored/@ark-ui/react/dist/components/number-input/number-input.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const numberInputContext = require('./number-input-context.cjs');
const numberInputControl = require('./number-input-control.cjs');
const numberInputDecrementTrigger = require('./number-input-decrement-trigger.cjs');
const numberInputIncrementTrigger = require('./number-input-increment-trigger.cjs');
const numberInputInput = require('./number-input-input.cjs');
const numberInputLabel = require('./number-input-label.cjs');
const numberInputRoot = require('./number-input-root.cjs');
const numberInputScrubber = require('./number-input-scrubber.cjs');



exports.Context = numberInputContext.NumberInputContext;
exports.Control = numberInputControl.NumberInputControl;
exports.DecrementTrigger = numberInputDecrementTrigger.NumberInputDecrementTrigger;
exports.IncrementTrigger = numberInputIncrementTrigger.NumberInputIncrementTrigger;
exports.Input = numberInputInput.NumberInputInput;
exports.Label = numberInputLabel.NumberInputLabel;
exports.Root = numberInputRoot.NumberInputRoot;
exports.Scrubber = numberInputScrubber.NumberInputScrubber;
