// @ark-ui/react@3.3.0/dist/components/combobox/use-combobox.js
'use client';
import * as combobox from '@zag-js/combobox';
import { useMachine, normalizeProps } from '@zag-js/react';
import { useMemo, useId } from 'react';
import { useEnvironmentContext } from '../../providers/environment/use-environment-context.js';
import { useLocaleContext } from '../../providers/locale/use-locale-context.js';
import { createSplitProps } from '../../utils/create-split-props.js';
import { useEvent } from '../../utils/use-event.js';

const useCombobox = (props) => {
  const [collectionOptions, comboboxProps] = createSplitProps()(props, [
    "isItemDisabled",
    "itemToValue",
    "itemToString",
    "items"
  ]);
  const collection = useMemo(
    () => combobox.collection(collectionOptions),
    Object.values(collectionOptions)
  );
  const { dir } = useLocaleContext();
  const { getRootNode } = useEnvironmentContext();
  const initialContext = {
    id: useId(),
    dir,
    getRootNode,
    collection,
    open: props.defaultOpen,
    value: props.defaultValue,
    "open.controlled": props.open !== void 0,
    ...comboboxProps
  };
  const context = {
    ...initialContext,
    collection,
    value: props.value,
    onValueChange: useEvent(props.onValueChange, { sync: true }),
    onInputValueChange: useEvent(props.onInputValueChange, { sync: true }),
    onHighlightChange: useEvent(props.onHighlightChange),
    onOpenChange: useEvent(props.onOpenChange)
  };
  const [state, send] = useMachine(combobox.machine(initialContext), {
    context
  });
  return combobox.connect(state, send, normalizeProps);
};

export { useCombobox };
