// @ark-ui/react/dist/components/date-picker/use-date-picker.js
'use client';
import * as datePicker from '@zag-js/date-picker';
import { useMachine, normalizeProps } from '@zag-js/react';
import { useId } from 'react';
import { useEnvironmentContext } from '../../providers/environment/use-environment-context.js';
import { useLocaleContext } from '../../providers/locale/use-locale-context.js';
import { useEvent } from '../../utils/use-event.js';

const useDatePicker = (props = {}) => {
  const { getRootNode } = useEnvironmentContext();
  const { dir } = useLocaleContext();
  const initialContext = {
    id: useId(),
    dir,
    getRootNode,
    open: props.defaultOpen,
    "open.controlled": props.open !== void 0,
    ...props,
    focusedValue: props.focusedValue ? datePicker.parse(props.focusedValue) : void 0,
    value: props.defaultValue ? datePicker.parse(props.defaultValue) : void 0,
    max: props.max ? datePicker.parse(props.max) : void 0,
    min: props.min ? datePicker.parse(props.min) : void 0
  };
  const context = {
    ...initialContext,
    max: props.max ? datePicker.parse(props.max) : void 0,
    min: props.min ? datePicker.parse(props.min) : void 0,
    value: props.value ? datePicker.parse(props.value) : void 0,
    onValueChange: useEvent(props.onValueChange, { sync: true }),
    onFocusChange: useEvent(props.onFocusChange),
    onViewChange: useEvent(props.onViewChange),
    onOpenChange: useEvent(props.onOpenChange)
  };
  const [state, send] = useMachine(datePicker.machine(initialContext), { context });
  return datePicker.connect(state, send, normalizeProps);
};

export { useDatePicker };
