// @ark-ui/react@3.3.0/dist/components/select/use-select.cjs
'use client';
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const react$1 = require('@zag-js/react');
const select = require('@zag-js/select');
const react = require('react');
const useEnvironmentContext = require('../../providers/environment/use-environment-context.cjs');
const useLocaleContext = require('../../providers/locale/use-locale-context.cjs');
const createSplitProps = require('../../utils/create-split-props.cjs');
const useEvent = require('../../utils/use-event.cjs');

function _interopNamespaceDefault(e) {
  const n = Object.create(null, { [Symbol.toStringTag]: { value: 'Module' } });
  if (e) {
    for (const k in e) {
      if (k !== 'default') {
        const d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: () => e[k]
        });
      }
    }
  }
  n.default = e;
  return Object.freeze(n);
}

const select__namespace = /*#__PURE__*/_interopNamespaceDefault(select);

const useSelect = (props) => {
  const [collectionOptions, selectProps] = createSplitProps.createSplitProps()(props, [
    "isItemDisabled",
    "itemToValue",
    "itemToString",
    "items"
  ]);
  const collection = react.useMemo(
    () => select__namespace.collection(collectionOptions),
    Object.values(collectionOptions)
  );
  const { getRootNode } = useEnvironmentContext.useEnvironmentContext();
  const { dir } = useLocaleContext.useLocaleContext();
  const initialContext = {
    id: react.useId(),
    dir,
    getRootNode,
    collection,
    open: props.defaultOpen,
    value: props.defaultValue,
    "open.controlled": props.open !== void 0,
    ...selectProps
  };
  const context = {
    ...initialContext,
    collection,
    value: props.value,
    onValueChange: useEvent.useEvent(props.onValueChange, { sync: true }),
    onHighlightChange: useEvent.useEvent(props.onHighlightChange),
    onOpenChange: useEvent.useEvent(props.onOpenChange)
  };
  const [state, send] = react$1.useMachine(select__namespace.machine(initialContext), {
    context
  });
  return select__namespace.connect(state, send, react$1.normalizeProps);
};

exports.useSelect = useSelect;
